src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedImplementationMethod.cs

```diff
@@ -97,7 +97,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
             if (returnType.ContainsNullableReferenceTypes())
             {
-                AddSynthesizedAttribute(ref attributes, compilation.SynthesizeNullableAttribute(returnType));
+                AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeNullableAttribute(this, returnType));
```

Comment:
>AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeNullableAttribute(this, returnType)); [](start = 16, length = 101)

Where does this attribute go? Shouldn't it be applied to a return type instead?

Reply:
Please add a unit-test that hits this code path.

---
In reply to: [147534168](https://github.com/dotnet/roslyn/pull/22820#discussion_r147534168) [](ancestors = 147534168)

Reply:
Added `EmitAttribute_ExplicitImplementationForwardingMethod`.

Reply:
> Added EmitAttribute_ExplicitImplementationForwardingMethod

It doesn't look like that test verifies the result of this method. I believe this method adds attributes on the method itself, rather than on the return type of the method, which is incorrect. I think we should simply remove this override of AddSynthesizedAttributes and open an issue to do the same in the "parent" branch.

Reply:
Logged #22952.